scripts/build-rpm.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
+set -eu -o pipefail
+
+source /etc/os-release
+
+distro="$ID"
+project_dir="$(cd $(dirname "${BASH_SOURCE[0]}")/.. && pwd)"
+version="$(git describe --tags | cut -f 1 -d "-" | sed 's/^v//')"
+release="dev_$(git describe --tags --dirty --always | sed 's/^v//g' | sed 's/-/_/g')"
+output_dir="${project_dir}/out"
+
+echo "project root: $project_dir"
+echo "build version: $version"
+echo "build release: $release"
+
+set -x
+
+# Create rpmbuild directory layout.
+build_dir="$(mktemp -d)"
+export HOME="$build_dir"
+mkdir -p "${build_dir}"/rpmbuild/{BUILD,RPMS,SOURCES,SPECS,SRPMS}
+
+# Azure Linux / Mariner requires rpmmacros.
+if [[ $ID == "mariner" || $ID == "azurelinux" ]]; then
+    echo '%_topdir %(echo $HOME)/rpmbuild' > "${build_dir}/.rpmmacros"
+fi
+
+# Create source tarball.
+output_source="${build_dir}/rpmbuild/SOURCES/azure-nvme-utils_dev.tgz"
+cd "$project_dir"
+git archive --verbose --format=tar.gz --prefix="azure-nvme-utils-$version/" HEAD --output "${output_source}"
+
+# Create spec file from template.
+cd "$project_dir/packaging/$distro"
+cat azure-nvme-utils.spec.tmpl | sed "s/__VERSION__/$version/g" | sed "s/__RELEASE__/$release/g" > azure-nvme-utils.spec
```

Comment:
If you decide to not go the template route this doesn't matter, but ShellCheck notes this could be written
```suggestion
sed "s/__VERSION__/$version/g" azure-nvme-utils.spec.tmpl | sed "s/__RELEASE__/$release/g" > azure-nvme-utils.spec
```

Reply:
updated, thanks!